lib/clas.gi

```diff
@@ -433,41 +433,34 @@ InstallGlobalFunction(ConjugacyClassesForSmallGroup,function(G)
   fi;
 end);
 
+InstallGlobalFunction( ConjugacyClassesForSolvableGroup,
+  function( G )
+  local   cls,  cl,  c;
 
-InstallMethod( ConjugacyClasses, "for groups: try random search",
+  cls := [  ];
+  for cl  in ClassesSolvableGroup( G, 0 )  do
+    c := ConjugacyClass( G, cl.representative, cl.centralizer );
+    Assert(2,Centralizer(G,cl.representative)=cl.centralizer);
+    Add( cls, c );
+  od;
+  Assert(1,Sum(cls,Size)=Size(G));
+  return cls;
+end );
+
+InstallMethod( ConjugacyClasses, "for groups: try solvable method and random search",
   [ IsGroup and IsFinite ],
 function(G)
 local cl;
   cl:=ConjugacyClassesForSmallGroup(G);
   if cl<>fail then
     return cl;
+  elif IsSolvableGroup( G ) and CanEasilyComputePcgs(G) then
```

Comment:
Here perhaps we should instead start with `HasIsSolvableGroup( G ) and ...`, as this code could be called on arbitrary groups. The only reason I didn't do it was that the existing method didn't do it either.

BTW, `CanEasilyComputePcgs` is a category, and set by some implications once the group "knows" that it is solvable. So the order of the check here is crucial.